bindings/java/src/lib.rs

```diff
@@ -15,20 +15,97 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use std::cell::RefCell;
 use std::collections::HashMap;
+use std::ffi::c_void;
 use std::str::FromStr;
+use std::sync::Arc;
 
-use jni::objects::JClass;
 use jni::objects::JMap;
 use jni::objects::JObject;
 use jni::objects::JString;
-use jni::sys::jboolean;
-use jni::sys::jlong;
-use jni::JNIEnv;
+use jni::objects::{JClass, JValue};
+use jni::sys::{jboolean, jint};
+use jni::sys::{jlong, JNI_VERSION_1_8};
+use jni::{JNIEnv, JavaVM};
+use once_cell::sync::OnceCell;
+use tokio::runtime::{Builder, Runtime};
+
 use opendal::BlockingOperator;
 use opendal::Operator;
 use opendal::Scheme;
 
+static mut RUNTIME: OnceCell<Runtime> = OnceCell::new();
+
+thread_local! {
+    static JAVA_VM: RefCell<Option<Arc<JavaVM>>> = RefCell::new(None);
+    static JENV: RefCell<Option<*mut jni::sys::JNIEnv>> = RefCell::new(None);
+}
+
+fn get_system_property(env: &mut JNIEnv, key: &str) -> Result<String, jni::errors::Error> {
+    let system_class = env.find_class("java/lang/System")?;
+    let key = env.new_string(key)?;
+    let value = env.call_static_method(
+        system_class,
+        "getProperty",
+        "(Ljava/lang/String;)Ljava/lang/String;",
+        &[JValue::Object(key.as_ref())],
+    )?;
+
+    Ok(env
+        .get_string(JString::from(value.l().unwrap()).as_ref())?
+        .into())
+}
+
+/// # Safety
+///
+/// This function could be only called by java vm when load this lib.
+#[no_mangle]
+pub unsafe extern "system" fn JNI_OnLoad(vm: JavaVM, _: *mut c_void) -> jint {
+    let mut env = vm.get_env().unwrap();
+    let thread_count = match get_system_property(&mut env, "opendal.thread.count") {
```

Comment:
I prefer to remove this to keep simple. We can make this configurable in the future.